Makefile

```diff
@@ -8,41 +7,114 @@ IMG ?= quay.io/${USERNAME}/${BASE_IMAGE}:${TAG}
+### --- P o s t g r e s --- ###
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
 
-start: ## Start all the components, compile & run (ensure goreman is installed, with 'go install github.com/mattn/goreman@latest')
-	$(GOBIN)/goreman start
+deploy-postgresql: ## Deploy postgres into Kubernetes and insert 'db-schema.sql'
+	kubectl create namespace gitops 2> /dev/null || true
+	kubectl -n gitops apply -f  $(MAKEFILE_ROOT)/manifests/postgresql-staging/postgresql-staging.yaml
+	$(MAKEFILE_ROOT)/create-dev-env.sh kube
 
-build: build-backend build-cluster-agent ## Build all the components - note: you do not need to do this before running start
+undeploy-postgresql: ## Undeploy postgres from Kubernetes
+	kubectl -n gitops delete -f  $(MAKEFILE_ROOT)/manifests/postgresql-staging/postgresql-staging.yaml
+
+### --- B a c k e n d --- ###
+# ~~~~~~~~~~~~~~~~~~~~~~~~~ #
+deploy-backend-crd: ## Deploy backend related CRDs
+	kubectl create namespace gitops 2> /dev/null || true
+	kubectl -n gitops apply -f  $(MAKEFILE_ROOT)/backend/config/crd/bases/managed-gitops.redhat.com_gitopsdeployments.yaml
+	kubectl -n gitops apply -f  $(MAKEFILE_ROOT)/backend/config/crd/bases/managed-gitops.redhat.com_gitopsdeploymentsyncruns.yaml
+
+undeploy-backend-crd: ## Remove backend related CRDs
+	kubectl -n gitops delete -f  $(MAKEFILE_ROOT)/backend/config/crd/bases/managed-gitops.redhat.com_gitopsdeployments.yaml
+	kubectl -n gitops delete -f  $(MAKEFILE_ROOT)/backend/config/crd/bases/managed-gitops.redhat.com_gitopsdeploymentsyncruns.yaml
+
+deploy-backend-rbac: ## Deploy backend related RBAC resouces
+	kubectl create namespace gitops 2> /dev/null || true
+	kubectl -n gitops apply -f  $(MAKEFILE_ROOT)/manifests/backend-rbac/
+
+undeploy-backend-rbac: ## Remove backend related RBAC resouces
+	kubectl delete -f  $(MAKEFILE_ROOT)/manifests/backend-rbac/
+
+deploy-backend: deploy-backend-crd deploy-backend-rbac ## Deploy backend operator into Kubernetes -- e.g. make deploy-backend IMG=quay.io/pgeorgia/gitops-service:latest
+	kubectl create namespace gitops 2> /dev/null || true
+	COMMON_IMAGE=${IMG} envsubst < $(MAKEFILE_ROOT)/manifests/managed-gitops-backend-deployment.yaml | kubectl apply -f -
+
+undeploy-backend: undeploy-backend-rbac undeploy-backend-crd ## Undeploy backend from Kubernetes
+	kubectl delete -f $(MAKEFILE_ROOT)/manifests/managed-gitops-backend-deployment.yaml
 
 build-backend: ## Build backend only
 	cd $(MAKEFILE_ROOT)/backend && make build
 
+test-backend: ## Run tests for backend only
+	cd $(MAKEFILE_ROOT)/backend && make test
+
+### --- c l u s t e r - a g e n t --- ###
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+
+deploy-cluster-agent-crd: ## Deploy cluster-agent related CRDs
+	kubectl create namespace gitops 2> /dev/null || true
+	kubectl -n gitops apply -f  $(MAKEFILE_ROOT)/backend-shared/config/crd/bases/managed-gitops.redhat.com_operations.yaml
+	kubectl create ns argocd 2> /dev/null || true
```

Comment:
Why maintain scripts/commands to install upstream Argo CD when you can install using the Operator on OpenShift?